common/beerocks/bwl/dummy/nl80211_client_dummy.cpp

```diff
@@ -0,0 +1,50 @@
+/* SPDX-License-Identifier: BSD-2-Clause-Patent
+ *
+ * Copyright (c) 2020 MaxLinear
+ *
+ * This code is subject to the terms of the BSD+Patent license.
+ * See LICENSE file for more details.
+ */
+
+#include "nl80211_client_dummy.h"
+
+namespace bwl {
+
+nl80211_client_dummy::nl80211_client_dummy() {}
+
+nl80211_client_dummy::~nl80211_client_dummy() {}
+
+bool nl80211_client_dummy::get_sta_info(const std::string &local_interface_name,
+                                        const sMacAddr &sta_mac_address, sNl80211StaInfo &sta_info)
+{
+    (void)local_interface_name;
+    (void)sta_mac_address;
```

Comment:
Using `__attribute__((unused))` is probably an even better idea (nitpick)

Reply:
Although we will never use another compiler, `__attribute__((unused))` is a proprietary GCC extension, so I'll stick to `(void)` and add a comment

Reply:
We'll only ever use GCC *or clang*. But both support `__attribute__((unused))`.

The typical way to handle this is something like

```
#if defined(__GCC__) || defined(__CLANG__)
#define UNUSED __attribute__((unused))
#else
#define UNUSED
#endif
```

However, for this particular case, it's about unused parameters, and we pass `-Wno-unused-parameter` in CMAKE_C_FLAGS, so why do we need this at all? Note BTW that `-Wunused-parameter` is a `-Wextra` warning because it is very common that parameters are unused.

Reply:
The reason we should still support it is that when prplMesh is built in-tree (OpenWRT, prplWrt, UGW, RDKB), these flags may be passed from the build system (unless I'm mistaken) - in which case I think we should be on the safe side and be strict with our CMAKE flags default settings.
I think the typical way of handling this @arnout suggested is a good compromise.